modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -1,24 +1,34 @@
-:navtitle: Adding a Visual Studio Code Extension
+:navtitle: Adding a Visual Studio Code extension to a workspace
 :keywords: Visual Studio Code extension, user-guide
 
 [id="adding-visual-studio-code-extension_{context}"]
-= Adding a Visual Studio Code Extension
+= Adding a Visual Studio Code extension to a workspace
 
-IDE specific plug-ins are no longer part of devfile with devfile v2. Instead, it uses the specific meta-folder to describe the plug-ins.
+Devfiles in the devfile v2 format don't contain any IDE-specific plug-ins. Instead, the plug-ins are described in a specific meta-folder.
 
-[id="visual-studio-code-extensions-json"]
-== Using `.vscode/extensions.json`
+[id="plug-in-registries"]
+== {prod-short} plug-in registries overview
 
-The easiest way to add a Visual Studio Code extension is to add the extension to the `.vscode/extensions.json` file. Using Che-Theia, the extension will be automatically installed and configured. With Che-Code, the Visual Studio Code fork, at the startup of the workspace, Visual Studio Code shows a recommendation pop-up to install the extension.
+{prod-short} has a default plug-in registry installed in every {prod-short} instance. The Che-Theia IDE searches and installs the extensions from that location. 
 
-.Procedure
+Check this {prod-short} https://github.com/eclipse-che/che-plugin-registry[registry project] for an overview of default plug-ins and source code. An online instance that is refreshed after every commit to the main branch, is located https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/[here]. You can use a different plug-in registry for Che-Theia if you don't work in air gap mode: only the default registry is available there.
+
+The plug-in overview for Che-Code Visual Studio Code editor is located in the https://www.open-vsx.org/[OpenVSX instance]. Air gap mode is not yet supported for this editor.
+
+[id="visual-studio-code-extensions-json"]
+== Adding an extension to .vscode/extensions.json
 
-. Add `.vscode/extensions.json` file if it is missing at the root of the Git repository.
+The easiest way to add a Visual Studio Code plug-in to a workspace is to add it to the `.vscode/extensions.json` file. The main advantage of this method is that it works with all supported {prod-short} IDEs.
 
-. Add the extension using its identifier. Publisher and extension names use a dot-separated format.
+If you use the Che-Theia IDE, the extension is installed and configured automatically. If you use a different supported IDE with the Che-Code Visual Studio Code fork, a pop-up with a recommendation to install the plug-in shows when you start the workspace.
 
-. Example of adding the Visual Studio Code Java extension:
+.Prerequisites
+. You have the `.vscode/extensions.json` file in the root of the GitHub repository.
+
+.Procedure
+. Add the plug-in id to `.vscode/extensions.json` file. Use a `.` sign to separate the publisher and plug-in names:
```

Comment:
Add the extension id to `.vscode/extensions.json` file. Use a `.` sign to separate the publisher and extension names: